### PR TITLE
:shirt: refactor: tweak API

### PR DESCRIPTION
**Pull request by kazupon**

### CODE REVIEW

File: src/Api.ts [hunk 1]
```diff
@@ -76,18 +75,20 @@ export default class VueI18NExtract {
...
-      reportString,
-      (err) => {
-        if (err) {
-          throw err;
-        }
-        // tslint:disable-next-line
-        console.log(`The report has been has been saved to ${writePath}`);
-      },
-    );
+    return new Promise((resolve, reject) => {
```
Comment (kazupon) [inline src/Api.ts]: We need return Promise for API user.

File: src/Api.ts [hunk 2]
```diff
@@ -76,18 +75,20 @@ export default class VueI18NExtract {
...
     const reportString = JSON.stringify(report);
-    return fs.writeFile(
-      path.resolve(process.cwd(), writePath),
-      reportString,
-      (err) => {
-        if (err) {
-          throw err;
-        }
-        // tslint:disable-next-line
-        console.log(`The report has been has been saved to ${writePath}`);
```
Comment (kazupon) [inline src/Api.ts]: We should be moved  to API call-side from API internal, It keeps on loosely coupled as an AP in doing so.

File: src/index.ts [hunk 3]
```diff
@@ -57,7 +56,9 @@ function report (command: any): void {
   api.logI18NReport(i18nReport);
 
   if (output) {
-    api.writeReportToFile(i18nReport, output);
+    await api.writeReportToFile(i18nReport, path.resolve(process.cwd(), output));
+    // tslint:disable-next-line
+    console.log(`The report has been has been saved to ${output}`);
```
Comment (kazupon) [inline src/index.ts]: ref: https://github.com/pixari/vue-i18n-extract/pull/44/files#r270988904

File: src/library/log-tables.ts [hunk 4]
```diff
@@ -36,3 +36,8 @@ export function logUnusedKeys (keys: I18NItem[]): void {
   // tslint:disable-next-line
   console.log(table.toString());
 }
+
+export default {
```
Comment (kazupon) [inline src/library/log-tables.ts]: export by the object to make it easier to unit test

**Comment by pixari** [comment]
Well done, @kazupon!

**Comment by kazupon** [comment]
Thanks!

**Comment by Spittal** [comment]
👍

**Pull request merged**